### PR TITLE
Added 5 LUTs to the multilut shader, based on Adobe Lightroom presets

### PR DESCRIPTION
**Pull request by FransBouma**
Added 4 LUTs to the multiLUT shader. The texture is now 227KB. I see the original one is optimized for web, no idea why, as 227KB or thereabout is small enough but anyway, this gives the merge conflict according to Github but git doesn't deal well with binaries so it's a replacement anyway. 

They look like this:
multi_lut

and are based on Adobe Lightroom presets. 

I did try an online optimizer, and using compression and lesser colors squeezes 140KB out of the LUT, but should this really be done? as the LUT texture is rather small already and as they're lookup tables I think reducing the # of colors isn't really helping the end result, so I left the texture atlas as is at 227KB. But it's your call :) 

(edit) I've added an additional one from GameVogue, not pictured above.

### CODE REVIEW

**Comment by FransBouma** [comment]
Currently a bug in the texture AND shader (forgot to update the ui max value, and the shader gives with color8 the same result as flat & green, which is bizarre. Looking into it).

**Comment by FransBouma** [comment]
Fixed. (LUT texture is now 40KB smaller as well, no idea why, there's more data in it (the previous one had 1 LUT duplicated). Oh well... ;)

**Comment by FransBouma** [comment]
Added an Azure Blue - Red dual tone LUT created by GameVogue

File: Shaders/MultiLUT.fx [hunk 1]
```diff
@@ -26,8 +26,8 @@
 
 uniform int fLUT_LutSelector < 
 	ui_type = "combo";
-	ui_min= 0; ui_max=15;
-	ui_items="Neutral\0Color1\0Color2\0Color3 (Blue oriented)\0Color4 (Hollywood)\0Color5\0Color6\0Color7\0Color8\0Cool light\0Flat & green\0Red lift matte\0Cross process\0Sepia\0\B&W mid constrast\0\B&W high contrast\0";
+	ui_min= 0; ui_max=16;
```
Comment (crosire) [inline Shaders/MultiLUT.fx]: These don't actually have any effect on combo boxes.

**Comment by LuisBurdallo** [comment]
Hi mate, Not entirely sure where you would my messages so I decided to comment here too. I also tried getting in touch in Reshade forums in the Multilut topic. I am having issues trying to make a 64x64 cells by 24 luts. The new Multilut loads correctly but selecting any of the luts will give me a weird kaleidoscope color mix and some bandings. Any idea?

**Comment by FransBouma** [comment]
@LuisBurdallo Make sure this line: https://github.com/crosire/reshade-shaders/blob/master/Shaders/MultiLUT.fx#L20 
reflects the # of LUTs that are in the texture (so the # or rows of 32x32pixel blocks)

**Comment by LuisBurdallo** [comment]
They are 24 and I had it in that line. The thing is I was trying to use rows of Luts of 4096X64 instead of 32 that's why I am having issue but I thought changing the first two numbers to this is would work:

#ifndef fLUT_TextureName
	#define fLUT_TextureName "MultiLut_Atlas2.png"
#endif
#ifndef fLUT_TileSizeXY
	#define fLUT_TileSizeXY 64
#endif
#ifndef fLUT_TileAmount
	#define fLUT_TileAmount 64
#endif
#ifndef fLUT_LutAmount
	#define fLUT_LutAmount 24
#endif

**Comment by LuisBurdallo** [comment]
OK got it working I was missing the Neutral one. Thanks for replying as it pushed me to get it solved.

**Comment by crosire** [comment]
Rename the technique. It's what ReShade uses to distinguish effects.

**Comment by LuisBurdallo** [comment]
Thanks. Will try that.

**Comment by LuisBurdallo** [comment]
For some reason when I load any of the instances on their own they work fine but if I load them all even tho their Technique, Effect name and Picture names are different only one of them work fine the other 2 don't calculate properly.

**Comment by FransBouma** [comment]
The texture name and sampler name have to be different as well, as they're shared among techniques/shades: https://github.com/crosire/reshade-shaders/blob/master/Shaders/MultiLUT.fx#L53

**Comment by LuisBurdallo** [comment]
Texture and Sampler name are taken from here? Names are different in all 3 instances. Weird thing is that 2 of them work fine with each other but when I activate 1 of them with any of the others the problem occurs. It works fine on its own.

#define fLUT_TextureName

**Comment by FransBouma** [comment]
fLUT_TextureName is the file the texture is in. It's loaded into a textureobject which is referenced as an identifier defined on line 53. It's sampled with the sampler defined below it. These objects are global scoped. So if you have two shaders with the same element names they give conflicts. But I realize it should give compile errors not runtime glitches, so the issue is something else (although it's good to use different texture/sampler identifier names)

I also assume #defines aren't locally scoped as the shader doesn't use a namespace. If you define a namespace in your own shader (so wrap all code within namespace *name* { }) it should be fine too.

**Comment by LuisBurdallo** [comment]
I managed to get them working but I think I have changed too much and changed every single instance in the entire code of fLUT to fLUT1 and MultiLUT to MultiLUT1. Is that Ok or should or can I do something simpler? Not sure if doing it in any other way could use fewer resources or something.

**Pull request merged**